File: q100-q199/q100-109/q102/q102.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
import queue

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        if root == None:
            return []
        level_que = queue.SimpleQueue()
        ans = []
        level_que.put(root)
        while not level_que.empty():
            cur_level_cnt = level_que.qsize()
            cur_level = []
            for i in range(0, cur_level_cnt):
                node = level_que.get()
                if node.left != None:
                    level_que.put(node.left)
                if node.right != None:
                    level_que.put(node.right)
                cur_level.append(node.val)
            ans.append(cur_level)
        return ans

File: Sequence/q100-q199/q100-109/q102/q102.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<TreeNode*> level_que;
        vector<vector<int>> ans;
        if (root == NULL) {
            return ans;
        }
        level_que.push(root);
        while (!level_que.empty()) {
            int cur_level_cnt = level_que.size();
            vector<int> cur_level;
            for (int i = 0; i < cur_level_cnt; ++i) {
                TreeNode* node = level_que.front();
                if (node->left != NULL) {
                    level_que.push(node->left);
                }
                if (node->right != NULL) {
                    level_que.push(node->right);
                }
                cur_level.push_back(node->val);
                level_que.pop();
            }
            ans.push_back(cur_level);
        }
        return ans;
    }
};
